tools/recetox_aplcms/main.R

```diff
@@ -0,0 +1,72 @@
+library(recetox.aplcms)
+
+save_extracted_features <- function(df, filename) {
+  df <- as.data.frame(df)
+  columns <- c("mz", "pos", "sd1", "sd2", "area")
+  arrow::write_parquet(df[columns], filename)
+}
+
+save_feature_sample_table <- function(df, filename) {
+  columns <- c("feature", "mz", "rt", "sample", "sample_rt", "sample_intensity")
+  arrow::write_parquet(df[columns], filename)
+}
+
+known_table_columns <- function() {
+  c("chemical_formula", "HMDB_ID", "KEGG_compound_ID", "mass", "ion.type",
+    "m.z", "Number_profiles_processed", "Percent_found", "mz_min", "mz_max",
+    "RT_mean", "RT_sd", "RT_min", "RT_max", "int_mean(log)", "int_sd(log)",
+    "int_min(log)", "int_max(log)")
+}
+
+save_known_table <- function(df, filename) {
+  columns <- known_table_columns()
+  arrow::write_parquet(df[columns], filename)
+}
+
+read_known_table <- function(filename) {
+  arrow::read_parquet(filename, col_select = known_table_columns())
+}
+
+save_pairing <- function(df, filename) {
+  write.table(df, filename, row.names = FALSE, col.names = c("new", "old"))
+}
+
+save_extracted_features_as_collection <- function(dfs, filenames) {
+  filenames <- tools::file_path_sans_ext(filenames)
+  filenames <- paste0("extracted-", filenames, ".parquet")

```

Comment:
Error: IOError: Failed to open local file 'extracted-/tmp/tmpu50kf5iy/files/9/9/c/dataset_99c28e91-c998-4c35-88c7-530f62f1b693.parquet'. Detail: [errno 2] No such file or directory

Reply:
this has to work on something else than the actual given filepath, for the sake of collection making we could start a a filename pattern and then use `<discover_datasets>`